runtime/src/iree/hal/drivers/cuda/dynamic_symbols.c

```diff
@@ -101,6 +101,46 @@ iree_status_t iree_hal_cuda_dynamic_symbols_initialize(
   return status;
 }
 
+static iree_status_t iree_hal_cuda_nccl_check_version(
+    iree_dynamic_library_t* nccl_library) {
+  IREE_ASSERT_ARGUMENT(nccl_library);
+
+  ncclResult_t (*ncclGetVersion)(int*) = NULL;
+
+  IREE_RETURN_IF_ERROR(iree_dynamic_library_lookup_symbol(
```

Comment:
As written, this will return IREE_STATUS_NOT_FOUND, I believe on a failure to find ncclGetVersion. I think this should be translated to IREE_STATUS_UNAVAILBLE. Since this function is unconditionally invoked even for single host cases, it basically means that a stray libnccl.so on the system (even if not at all what we are looking for) keeps the whole system from starting.

I think the better behavior is that if we fail the version check in any way, it means that the library is not available and startup proceeds without it.

Reply:
agree that we should have UNAVAILABLE consistently returned